src/main/java/org/broadinstitute/hellbender/tools/walkers/qc/CheckPileup.java

```diff
@@ -0,0 +1,226 @@
+    private PrintStream out;
+    /**
+     * By default the program will quit if it encounters an error (such as missing truth data for a given position).
+     * Use this flag to override the default behavior; the program will then simply print an error message and move on
+     * to the next position.
+     */
+    @Argument(fullName="continue_after_error",doc="Continue after encountering an error",optional=true)
+    public boolean continueAfterAnError = false;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    public long nLoci = 0;
+    public long nBases = 0;
+
+    public void apply(AlignmentContext context, ReferenceContext ref, FeatureContext tracker) {
+        ReadPileup pileup = context.getBasePileup();
+        SAMPileupFeature truePileup = getTruePileup( tracker );
+
+        if ( truePileup == null ) {
+            out.printf("No truth pileup data available at %s%n", pileup.getPileupString((char) ref.getBase()));
+            if ( !continueAfterAnError) {
+                throw new UserException.BadInput(String.format("No pileup data available at %s given GATK's output of %s -- this walker requires samtools pileup data over all bases",
+                        context.getLocation(), new String(pileup.getBases())));
+            }
+        } else {
+            String pileupDiff = pileupDiff(pileup, truePileup, true);
+            if ( pileupDiff != null ) {
+                out.printf("%s vs. %s%n", pileup.getPileupString((char) ref.getBase()), truePileup.getPileupString());
+                if ( !continueAfterAnError) {
+                    throw new UserException.BadInput(String.format("The input pileup doesn't match the GATK's internal pileup: %s", pileupDiff));
+                }
+            }
+        }
+
+        nLoci++;
+        nBases += pileup.size();
+    }
+
+    private static String maybeSorted( final String x, boolean sortMe )
+    {
+        if ( sortMe ) {
+            byte[] bytes = x.getBytes();
+            Arrays.sort(bytes);
+            return new String(bytes);
+        }
+        return x;
+    }
+
+    public String pileupDiff(final ReadPileup a, final SAMPileupFeature b, boolean orderDependent)
+    {
+        if ( a.size() != b.size() ) {
+            return "Sizes not equal";
+        }
+        if (IntervalUtils.compareLocatables(a.getLocation(), b, getReferenceDictionary()) != 0 ) {
+            return "Locations not equal";
+        }
+
+        String aBases = maybeSorted(new String(a.getBases()), ! orderDependent );
+        String bBases = maybeSorted(b.getBasesAsString(), ! orderDependent );
+        if ( ! aBases.toUpperCase().equals(bBases.toUpperCase()) ) {
+            return "Bases not equal";
+        }
+
+        String aQuals = maybeSorted(new String(a.getBaseQuals()), ! orderDependent );
```

Comment:
This has the strange property that if these were sorted the quals can be sorted differently than the bases.  It seems like they should be sorted together because each qual goes with a specific base.